pkg/osd_cluster_upgrader/upgrader.go

```diff
@@ -170,10 +170,36 @@ func CommenceUpgrade(c client.Client, cfg *osdUpgradeConfig, scaler scaler.Scale
 	if err != nil {
 		return false, err
 	}
+
+	// Move the cluster to the same channel first
+	if cv.Spec.Channel != desired.Channel {
+		logger.Info(fmt.Sprintf("Moving cluster from Channel %s to Channel %s", cv.Spec.Channel, desired.Channel))
+		cv.Spec.Channel = upgradeConfig.Spec.Desired.Channel
```

Comment:
Keep to using `desired.Channel` since we check (`cv.Spec.Channel != desired.Channel`) against that version

Reply:
Good spot, fixed in 6539f2c